relayer/chains/cosmos/feegrant.go

```diff
@@ -274,19 +313,21 @@ func (cc *CosmosProvider) EnsureBasicGrants(ctx context.Context, memo string) (*
 		hasGrant := false
 		for _, basicGrant := range validGrants {
 			if basicGrant.Grantee == granteeAddr {
-				fmt.Printf("Valid grant found for granter %s, grantee %s\n", basicGrant.Granter, basicGrant.Grantee)
 				hasGrant = true
 			}
 		}
 
-		if !hasGrant {
-			grantsNeeded += 1
-			fmt.Printf("Grant will be created on chain for granter %s and grantee %s\n", granterAddr, granteeAddr)
+		if !hasGrant && !cc.PCfg.FeeGrants.IsExternalGranter {
+			grantsNeeded++
+			cc.log.Info("Creating feegrant", zap.String("granter", granterAddr), zap.String("grantee", granteeAddr))
+
 			grantMsg, err := cc.getMsgGrantBasicAllowance(granterAcc, granteeAcc)
 			if err != nil {
 				return nil, err
 			}
 			msgs = append(msgs, grantMsg)
+		} else if !hasGrant {
+			cc.log.Warn("Missing feegrant", zap.String("external granter", granterAddr), zap.String("grantee", granteeAddr))
```

Comment:
nit: log field names should be snake_case https://github.com/cosmos/relayer/blob/main/docs/logging.md#usage